lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -3348,9 +3353,15 @@ def show_options(mod) # :nodoc:
     mod_opt = Serializer::ReadableText.dump_options(mod, '   ')
     print("\nModule options (#{mod.fullname}):\n\n#{mod_opt}\n") if (mod_opt and mod_opt.length > 0)
 
-    # If it's an exploit and a payload is defined, create it and
-    # display the payload's options
-    if (mod.exploit? and mod.datastore['PAYLOAD'])
+    # We have to special-case browser autopwn because BAP is an exploit module that allows having
+    # multiple payloads, but normally MSF can't do this, so it will have be handled by the BAP
+    # mixin.
+    # For other normal cases, if it's still an exploit and a payload is defined, then just go ahead
+    # create it, and then display the payload options.
+    if mod.exploit? and mod.kind_of?(Msf::Exploit::Remote::BrowserAutopwnv2) and mod.respond_to?(:show_payloads)
```

Comment:
Same thing as above, we should be able to avoid this hackery by moving to an auxiliary and coming up with smart payload selection options.